doc/classes/ProjectSettings.xml

```diff
@@ -561,10 +561,10 @@
 			When set to [code]warn[/code] or [code]error[/code], produces a warning or an error respectively when defining a local or member variable, signal, or enum that would have the same name as a built-in function or global class name, thus shadowing it.
 		</member>
 		<member name="debug/gdscript/warnings/shadowed_variable" type="int" setter="" getter="" default="1">
-			When set to [code]warn[/code] or [code]error[/code], produces a warning or an error respectively when defining a local or member variable that would shadow a member variable that the class defines.
+			When set to [code]warn[/code] or [code]error[/code], produces a warning or an error respectively when defining a local or class variable that would shadow a member that the class defines.
```

Comment:
This warning doesn't seem to trigger if an enum member shadows instead what seemed to happen before this PR is that an error was pushed saying the member already exists in a parent class. If this is intentional, should I change the doc to the above but remove the enum member part?

Reply:
Judging by the code, this is supposed to work:

https://github.com/godotengine/godot/blob/87318a2fb7fffeb72adca934e31915be077c3d1f/modules/gdscript/gdscript_analyzer.cpp#L1145-L1148

If for some reason it doesn't work, we should fix it. If you want to do it, you can fix it within this PR.

Reply:
I believe it is impossible to reach line 1147 because the parser would parse elements of an unnamed enum as enum_values so I don't know in what case `member.m_enum->identifier->name == StringName()` could be true.

Can we clarify here? By enum members do we mean members that are enums or enum values? Because the comment and code seems to try to see if enum values are shadowing a member.

Reply:
Unfortunately, we don't have a unified naming for enum elements. We call them "constants", "enum values", "enum members". Also note that there are unnamed enums, their elements fall through to the outer scope.

Reply:
"[EnumName] constants" is the most consistent in the class reference and is what should be used.

Reply:
I am wondering whether if would be possible to reach line 1147 because to reach the condition. 
`member.type` would have to be `GDScriptParser::ClassNode::Member::ENUM`. However if the parser was parsing an unnamed enum then this case block would never be entered. Instead the case block for `GDScriptParser::ClassNode::Member::ENUM_VALUE` would be entered for elements of the unnamed enum. I have confirmed this through the editor by logging messages once inside the different cases.

I may be misunderstanding here and I would appreciate if you could show me a time when line 1147 is reached.

So I am wondering if it makes sense for the shadowing check to be moved to the case for `GDScriptParser::ClassNode::Member::ENUM_VALUE`. However, if it true that the check mentioned never happened before then I wonder if this warning is redundant due to the fact that we check for member name conflict in and this pushes an error:
```
case GDScriptParser::ClassNode::Member::ENUM_VALUE: {
	member.enum_value.identifier->set_datatype(resolving_datatype);

	if (member.enum_value.custom_value) {
		check_class_member_name_conflict(p_class, member.enum_value.identifier->name, member.enum_value.custom_value);
```

Reply:
Probably the warning for unnamed enum members is not needed at all. Sorry for the confusion.
![](https://github.com/user-attachments/assets/233a52dd-e606-4394-86aa-d6750225fed0)

Reply:
Is there something we can fix for this issue regarding SHADOWED_GLOBAL_INDENTIFIER or should we open an issue specifically for that once we have some reproducible behaviour?

Reply:
You are right, `SHADOWED_VARIABLE` and `SHADOWED_VARIABLE_BASE_CLASS` will never be produced for enum members, so it shouldn't be in the documentation. As for `SHADOWED_GLOBAL_IDENTIFIER`, we could fix that in a separate PR, as you see fit.

Reply:
I updated the docs to mentioned local variables and constants but not enums.